mlir/lib/Target/LLVMIR/ModuleTranslation.cpp

```diff
@@ -641,11 +748,15 @@ ModuleTranslation::convertOperation(Operation &op,
                         "dialect for op: ")
            << op.getName();
 
+  InstructionCapturingInserter::CollectionScope scope(
+      builder, nestedConvertOperationUsesCapturingBuilder);
+  CapturingBuilderFlagScope flagScope(*this, false);
```

Comment:
Would it be possible to pass the isCapturingBuilder flag together with the builder into convertOperation/convertBlock? It seems the flag is only true if the convertOperation is directly called from convertOneFunction? I assume the problem is that these are public functions and adding private impl functions makes things also quite complex.

Overall the flag scope solution is fine. I was just wondering if there is a more explicit approach.